src/compiler/factory.ts

```diff
@@ -3016,6 +3032,37 @@ namespace ts {
         );
     }
 
+    export function createExpressionForJsxFragment(jsxFactoryEntity: EntityName, reactNamespace: string, children: Expression[], parentElement: JsxOpeningFragment, location: TextRange): LeftHandSideExpression {
+        const tagName = createPropertyAccess(
+            createReactNamespace(reactNamespace, parentElement),
+            "Fragment"
+        );

```

Comment:
Should this perhaps be customizable, similar to how `--jsxFactory` is? e.g. `--jsxFragmentComponent`?

Reply:
I agree with your second point here; to take the example of Mithril, the framework supports fragments, but with a completely different approach than React will (i.e. `m.fragment(attrs, children)`) so it's not as convenient as simply replacing the call to `createElement` with `--jsxFactory`. It seems to make sense in this case to just preserve the JSX syntax as is for non-React libs and have each define their own transpilation depending on their implementation of fragments.

Reply:
@Kovensky do you know what babel will do here? would it support another version of the paragma for the `React.createElement`?

Reply:
checked with @hzoo over slack and does not seem that babel have plans for this in the immediate term. 
I wounder if we should just make it an error to use Fragment + `--jsx React` + `--jsxNamespace`

Reply:
@uniqueiniquity can you file an issue for adding the error.

Reply:
yeah. that is why we are adding an error. this will ensure that we do not have any unintended uses of the Fragment element. once we add a new option we can remove/relax the error.

Reply:
Why would we want to error here? Seems like you're not helping that many people by preventing this.

Reply:
The concern was those that use --jsx react and --jsxFactory to emit for a different framework. In particular, Mithril has already supported [fragments](https://mithril.js.org/fragment.html) for about a year with the syntax `m.fragment(attrs, children)`, which doesn't really work with our current setup.

Reply:
True, but if you're importing React under a different name, who are we to stop users? Additionally, @developit @KFlash @trueadm, do Preact and Inferno intend to support fragments with a similar top-level API?

Reply:
> True, but if you're importing React under a different name, who are we to stop users?

it is not going to work.. the emit still emits `React.Fragment` in the otuput. so if `React` is not defined this is going to fail at runtime.
e.g. `--jsxFactory h`  results in `h(React.Fragment, null, [...])`


The question is how do we support this for non-React, non-`--reactNamesace`-users? one option is to add a new `--jsxFragmentParagma` or similar that defaults to `React.Fragment`.

Reply:
> it is not going to work.. 

I don't understand, the emit currently uses `createReactNamespace`.

Reply:
Oh, I think I see now. I'm confusing `jsxFactory` and `reactNamespace`

Reply:
So are you on board with issuing the error?

Reply:
Probably, but feel free not to block this PR on it.

Reply:
Updated the PR to issue an error.

Reply:
In the Babel PR, I add a pragma for fragments, as @mhegazy suggested. https://github.com/babel/babel/pull/6552/files#diff-7e5b807d66ce513f501563ad8f8d03e6R117
https://github.com/babel/babel/pull/6552/files#diff-7e5b807d66ce513f501563ad8f8d03e6R89